coalib/parsing/DefaultArgParser.py

```diff
@@ -72,6 +72,12 @@ def default_arg_parser(formatter_class=None):
                             help='Files that will be analyzed will be '
                                  'restricted to those in the globs listed '
                                  'in this argument as well the files setting')
+    arg_parser.add_argument('--caching',
```

Comment:
Are you sure you want to `--caching=flush` and `--caching` (they're not too related actually)? Help menu would be more confusing IMO. My proposal is `--caching` to enable caching, and `--rebuild-cache` to rebuild cache.

Also large number of args is pretty common :P Take a look at `ls --help`.